pandas/plotting/_matplotlib/boxplot.py

```diff
@@ -226,13 +226,24 @@ def boxplot(data, column=None, by=None, ax=None, fontsize=None,
     def _get_colors():
         #  num_colors=3 is required as method maybe_color_bp takes the colors
         #  in positions 0 and 2.
-        return _get_standard_colors(color=kwds.get('color'), num_colors=3)
+        #  if colors not provided, use color 2 for medians and 0 for all else
+        result = _get_standard_colors(num_colors=3)
+        result = np.take(result, [0, 0, 2, 0])
+
+        colors = kwds.pop('color', None)
+        if colors and isinstance(colors, dict):
+            valid_keys = ['boxes', 'whiskers', 'medians', 'caps']
+            for i in range(4):
+                if valid_keys[i] in colors:
+                    result[i] = colors[valid_keys[i]]
```

Comment:
Should invalid colors raise? To catch issues like `.boxplot(colors={'boxse': 'red'})`

Reply:
They did in `DataFrame.plot.box`, so yes, I think so